src/queue.ts

```diff
@@ -53,16 +53,23 @@ export class Queue extends EventEmitter {
       }
       // grab the element at the front of the array
       const item = this.q.shift()!;
+      // Depending on CPU load and other factors setTimeout() is not guranteed to run exactly
```

Comment:
`setTimeout`'s timing is not guaranteed, I was bumping into an issue in tests where it executed a few moments before `item.timeToRun`, making it so tests never executed.

A better fix would be to have a long running `setInterval` which ensures the queue is eventually drained. This felt like a more significant change that should happen outside of this PR.